### PR TITLE
Feature/support go chapter stack queue

### PR DESCRIPTION
**Pull request by Reanon**
1. Implement stack & queue in array.
2. Implement stack & queue in linkedlist.
3.Add testcase & benchmark in stack & queue.

### CODE REVIEW

File: codes/go/chapter_stack_and_queue/linkedlist_queue.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+package chapter_stack_and_queue
+
+import (
+	"container/list"
+	"fmt"
+	"strings"
+)
+
+// LinkedListQueue 基于链表实现的栈, 使用内置包 list 来实现栈
```
Comment (krahets) [inline codes/go/chapter_stack_and_queue/linkedlist_queue.go:13]: 应为“队列“

File: codes/go/chapter_stack_and_queue/stack_test.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
+// File: stack_test.go
+// Created Time: 2022-11-28
+// Author: Reanon (793584285@qq.com)
+
+package chapter_stack_and_queue
+
+import "testing"
+
```
Comment (krahets) [inline codes/go/chapter_stack_and_queue/stack_test.go:8]: Testing for the Go built-in stack is required.

File: codes/go/chapter_stack_and_queue/queue_test.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
+// File: queue_test.go
+// Created Time: 2022-11-28
+// Author: Reanon (793584285@qq.com)
+
+package chapter_stack_and_queue
+
+import "testing"
+
```
Comment (krahets) [inline codes/go/chapter_stack_and_queue/queue_test.go:8]: Testing for the Go built-in queue is required.

File: codes/go/chapter_stack_and_queue/deque.go [hunk 4]
```diff
@@ -0,0 +1,27 @@
+// File: deque.go
+// Created Time: 2022-11-29
+// Author: Reanon (793584285@qq.com)
+
+package chapter_stack_and_queue
+
+type Deque interface {
```
Comment (krahets) [inline codes/go/chapter_stack_and_queue/deque.go:7]: Testing for the Go built-in deque is required.

Comment (krahets) [reply codes/go/chapter_stack_and_queue/deque.go:7]: Do you use `interface` to reveal the usage of built-in Stack, Queue, and Deque?
I'm wondering if it is the only way as It is not consistent with other languages.

File: codes/go/chapter_stack_and_queue/linkedlist_queue.go (hunk 1 above)
Comment (Reanon) [reply codes/go/chapter_stack_and_queue/linkedlist_queue.go:13]: Fixed in next commit.

File: codes/go/chapter_stack_and_queue/stack_test.go (hunk 2 above)
Comment (Reanon) [reply codes/go/chapter_stack_and_queue/stack_test.go:8]: In actual, there is't a built-in stack in Go. So, i try to define a Go Interface to be aligned with Java code.

**Review by krahets — APPROVED** [review]
Thanks, @Reanon ! Now merge it.

**Pull request merged**